src/solver.rs

```diff
@@ -1383,6 +1491,39 @@ fn into_resized(vec: CsVec, len: usize) -> CsVec {
     CsVec::new(len, indices, data)
 }
 
+fn choose_branch_var(cur_solution: &Solution, domain: &[VarDomain]) -> Option<Variable> {
+    let mut max_divergence = 0.0;
+    let mut max_var = None;
+    for (var, &val) in cur_solution {
+        if domain[var.0] != VarDomain::Integer {
+            continue;
+        }
+        let divergence = f64::abs(val - val.round());
+        if divergence > 1e-5 && divergence > max_divergence {
```

Comment:
@Specy I think you should use the EPS constant here, instead of hardcoding 1e-5.